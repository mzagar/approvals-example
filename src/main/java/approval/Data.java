package approval;

import java.util.Map;

/**
 * Created with IntelliJ IDEA.
 * User: mzagar
 * Date: 4/15/13
 * Time: 10:04 PM
 * To change this template use File | Settings | File Templates.
 */
public class Data {
	private int id;
	private String text;
	private double value;
	private Map<Integer, String> map;

	public Data(int id, String text, double value, Map<Integer, String> map) {
		this.id = id;
		this.text = text;
		this.value = value;
		this.map = map;
	}

	public int getId() {
		return id;
	}

	public String getText() {
		return text;
	}

	public double getValue() {
		return value;
	}

	public Map<Integer, String> getMap() {
		return map;
	}
}
