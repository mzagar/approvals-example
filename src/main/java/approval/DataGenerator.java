package approval;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicInteger;

/**
 * Created with IntelliJ IDEA.
 * User: mzagar
 * Date: 4/15/13
 * Time: 10:04 PM
 * To change this template use File | Settings | File Templates.
 */
public class DataGenerator {

	private static final AtomicInteger id = new AtomicInteger(0);

	public Data generate() {
		return generate(1);
	}

	public Data generate(int factor) {
		int id = DataGenerator.id.incrementAndGet();
		String str = "random" + id;
		double dbl = (id*factor+1.234) / 3.14;
		Map<Integer, String> map = new HashMap<Integer, String>();
		map.put(id*100, "string: " + str);
		map.put(id*99, "double: " + dbl);
		map.put(id*98, "id: " + id);
		return new Data(id, str, dbl, map);
	}
}
