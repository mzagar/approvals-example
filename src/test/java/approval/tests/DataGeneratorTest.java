package approval.tests;

import approval.Data;
import approval.DataGenerator;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.*;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;
import util.ObjectMapperFactory;


import java.util.Date;

@UseReporter(JunitReporter.class)
public class DataGeneratorTest {

	private static ObjectMapper mapper = new ObjectMapperFactory().build();

	@Test
	public void test_string() throws Exception {
		String actual = new Date(1000L).toString();
		Approvals.verify(actual);
	}

	@Test
	public void returns_data() throws Exception {
		Data data = new DataGenerator().generate(1);
		Approvals.verify(json(data));
	}

	private static String json(Object o) {
		try {
			return mapper.writeValueAsString(o);
		} catch (JsonProcessingException e) {
			throw new RuntimeException(e.getMessage(),e);
		}
	}
}
