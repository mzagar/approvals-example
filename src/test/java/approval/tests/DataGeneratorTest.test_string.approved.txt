Thu Jan 01 01:00:01 CET 1970